src/vdaf.rs

```diff
@@ -219,12 +219,15 @@ where
 
 /// A state transition of an Aggregator during the Prepare process.
 #[derive(Debug)]
-pub enum PrepareTransition<S, M, O> {
+pub enum PrepareTransition<V: Aggregator>
+where
+    for<'a> &'a V::AggregateShare: Into<Vec<u8>>,
```

Comment:
If we're going to put this trait bound in `libprio`, could we put it right on `vdaf::Aggregator`? I think then we could delete corresponding trait bounds from quite a few places in Janus.

Reply:
It is on `vdaf::Aggregator` [L161 of this commit]; unfortunately, Rust makes this trait bound "infectious", rather than having the `V: Aggregator` bound itself imply the required bound on `&V::AggregateShare` in this case.

I thought GATs might simplify this, but I hopped into nightly temporarily to try them out and in some cases the bound is still infectious. It did allow eliding the bound in some other cases, though.

Unfortunately, I think this is a gap in Rust's generics currently. We could work around it by factoring the bound differently: instead of `Into<_>` on `&AggregateShare`, create a new bound named something like `Serializable`, specify it on `AggregateShare` per normal, and make `Serializable::serialize` take `&self` rather than `self`. (that is, the problem AFAICT is that Rust's support for expressing bounds on _references_ to associated types is not currently great)